src/shared/clients/devenvClient.ts

```diff
@@ -103,6 +108,72 @@ export class DevEnvClient implements vscode.Disposable {
         // `Authorization` _should_ have two parameters (RFC 7235), MDE should probably fix that
         headers: { Authorization: this.authToken },
     })
+
+    #lastKnownActivityTimestamp: number | undefined
+
+    async updateUserActivity(timestamp: number = Date.now()): Promise<number> {
+        if (this.isCodeCatalystDevEnv()) {
+            await this.got.put('activity', { json: { timestamp: timestamp.toString() } })
+        }
+        this.#lastKnownActivityTimestamp = timestamp
+        this.#updatedActivityEventEmitter.fire(timestamp)
+        return timestamp
+    }
+
+    async getUserActivity(): Promise<number | undefined> {
+        const response = await this.got<GetActivityResponse>('activity')
+        const latestTimestamp = response.body.timestamp ? parseInt(response.body.timestamp) : undefined
+
+        // There is a newer activity timestamp that this client did not know of
```

Comment:
A Dev Env can have multiple clients connected to it, so if one is active it will update the timestamp, and later if another one calls this get method, it will be different from what it currently has.

I'll update the comment to explain 'why'.